### PR TITLE
Add specimen and taxon circumscription

### PR DESCRIPTION
**Pull request by gaurav**
This PR incorporates two phyloreferences from [Fisher *et al*, 2007](https://bioone.org/journals/the-bryologist/volume-110/issue-1/0007-2745(2007)110%5b46%3aPOTCWA%5d2.0.CO%3b2/Phylogeny-of-the-Calymperaceae-with-a-rank-free-systematic-treatment/10.1639/0007-2745(2007)110[46:POTCWA]2.0.CO;2.short), which include examples of specimens and a taxon circumscribed by a publication, as well as one of the phylogenies from that paper for testing. In doing this, I found a bug with how phylorefs with external specifiers were being generated when there are multiple external specifiers, and rewrote the algorithm to eliminate this bug. I also added a example file `example_five_external_specifiers.json`, which has a phyloreference with five external specifiers, just to make sure those logical expressions are being generated correctly.

It also includes a minor fix to error reporting in `phyx2owl.js`.

Should be merged after PR #79 and PR #72.

### CODE REVIEW

**Comment by hlapp** [comment]
@gaurav dd you forget to mark this one ready for review, or did you find you have to return it to draft stage temporarily?

**Comment by gaurav** [comment]
@hlapp Oops, yes, I forgot to mark it ready for review! My bad.

File: src/wrappers/PhylorefWrapper.js [hunk 1]
```diff
@@ -784,24 +650,54 @@ class PhylorefWrapper {
...
+
+        // Collect all of the externals that are not selected.
+        const remainingExternals = externalSpecifiers.filter(ex => ex !== selectedExternal);
+
+        // Add the remaining externals, which we assume will resolve outside of
+        // this clade.
+        remainingExternals.forEach((externalTU) => {
+          intersectionExprs.push({
+            '@type': 'owl:Restriction',
+            onProperty: 'obo:CDAO_0000144', // has_Ancestor
```
Comment (hlapp) [inline src/wrappers/PhylorefWrapper.js:684]: Note that there's a property `has_outside_TU` for this in the next release of the Phyloref Ontology: https://github.com/phyloref/phyloref-ontology/blob/master/phyloref.ofn#L38

Not sure whether using it should be a downstream change.

**Review by hlapp — APPROVED** [review]
Look OK. See inline comment. And you still haven't explained to me (or maybe you have and I don't remember where?) why we are using the development version of the Phyx context.

File: src/wrappers/PhylorefWrapper.js (hunk 1 above)
Comment (gaurav) [reply src/wrappers/PhylorefWrapper.js:684]: Ah, I forgot about that property! I've opened an issue to make the change in Phyx.js once the new version of the Phyloref Ontology has been released: https://github.com/phyloref/phyx.js/issues/83.

**Comment by gaurav** [comment]
> Look OK. See inline comment.

Thanks, Hilmar!

> And you still haven't explained to me (or maybe you have and I don't remember where?) why we are using the development version of the Phyx context.

I tried to answer this question in https://github.com/phyloref/phyx.js/pull/72#discussion_r589107310 after you asked it here, but if you'd like me to document this in a README file in the `test/` directory, let me know!

**Comment by hlapp** [comment]
> > And you still haven't explained to me (or maybe you have and I don't remember where?) why we are using the development version of the Phyx context.
> 
> I tried to answer this question in [#72 (comment)](https://github.com/phyloref/phyx.js/pull/72#discussion_r589107310) after you asked it here, but if you'd like me to document this in a README file in the `test/` directory, let me know!

I think it would indeed be good to have that documented somewhere. That the question has already come up repeatedly indicates that before too long it will again, and so it seems useful to write this down so we don't have to dig around for the answer.

**Comment by gaurav** [comment]
> I think it would indeed be good to have that documented somewhere. That the question has already come up repeatedly indicates that before too long it will again, and so it seems useful to write this down so we don't have to dig around for the answer.

Sounds good! I've written them up in PR #84.

**Pull request merged**